test/test.js

```diff
@@ -62,6 +62,27 @@ async function run () {
     t.equal(resultA.body, 'this is a')
   })
 
+  test('basic proxy with query search', async t => {
+    const server = Fastify()
+    const querySearch = '?param=value'
+    server.register(proxy, {
+      upstream: `http://localhost:${origin.server.address().port}`
+    })
+
+    await server.listen(0)
+    t.teardown(server.close.bind(server))
+
+    const resultRoot = await got(
+      `http://localhost:${server.server.address().port}${querySearch}`
+    )
+    t.equal(resultRoot.body, 'this is root')
+
+    const resultA = await got(
+      `http://localhost:${server.server.address().port}/a${querySearch}`
+    )
+    t.equal(resultA.body, 'this is a')
```

Comment:
Can you a new route for `origin` that return the `request.raw.url` then assert it here to see if the `query` part is actually removed?
It seems like this test will not fail without your changes.

Reply:
Those tests pass even without my fix so I don't know what is going here that's why I asked to check it. If you have knowledge about it please check it :)